From c537ba23021df9334f8ba4adbeb6a72cb88f3182 Mon Sep 17 00:00:00 2001
From: Isaac Lee <isaac.lee@mediatek.com>
Date: Thu, 26 Jul 2018 17:07:25 +0800
Subject: [PATCH 6/6] Show bootanimation after decrypt

Because the original modification (restart SF which is added
for display bootanimation) from O to P cause bootanimation NE,
 we remove the part of restart SF and add other flow to show
 bootanimation.

Test: maunal, ran the test 10 times and it cause no NE and
      display BootAnimation aftert decrypt
Test: boot aosp_sailfish

Bug: 79547653
Change-Id: I355ccdbb2e2f27d897e2e0ee00f9300ef38ede03
---
 rootdir/init.rc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index eb7a0abe8..970b3f8d0 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -705,12 +705,12 @@ on property:vold.decrypt=trigger_restart_min_framework
     class_start main
 
 on property:vold.decrypt=trigger_restart_framework
-    stop surfaceflinger
-    start surfaceflinger
     # A/B update verifier that marks a successful boot.
     exec_start update_verifier
     class_start main
     class_start late_start
+    setprop service.bootanim.exit 0
+    start bootanim
 
 on property:vold.decrypt=trigger_shutdown_framework
     class_reset late_start
-- 
2.17.1

